packages/airnode-validator/src/config/config.ts

```diff
@@ -43,13 +43,29 @@ export const priorityFeeSchema = z.object({
     .optional(),
 });
 
-export const chainOptionsSchema = z.object({
-  txType: z.union([z.literal('legacy'), z.literal('eip1559')]),
-  gasPriceMultiplier: z.number().optional(),
-  baseFeeMultiplier: z.number().int().optional(),
-  priorityFee: priorityFeeSchema.optional(),
-  fulfillmentGasLimit: z.number().int(),
-});
+export const chainOptionsSchema = z
```

Comment:
You can simplify this using the https://github.com/colinhacks/zod#discriminated-unions

Also, the inferred TS types will be better.

Reply:
"simplify" 😅

I implemented using `discriminatedUnion`, which then required `strict()` validation in order to error on unrecognized  fields e.g. `priorityFee` being present for `legacy` `txType`. By default this produces a somewhat confusing error "Unrecognized key(s) in object", which I modified though a custom error mapping so that the user would know that the key(s) were unrecognized / disallowed _**for that transaction type**_.

Reply:
I think the error message from zod is generic because it uses the `path` to tell you what is wrong. I like your message better, but I wouldn't be so strict about it in general.